src/plugins/data/public/plugin.ts

```diff
@@ -18,14 +18,22 @@
  */
 
 import { PluginInitializerContext, CoreSetup, CoreStart, Plugin } from '../../../core/public';
+import { AutocompleteProviderRegister } from '../common/autocomplete_provider';
 
 export class DataPublicPlugin implements Plugin<{}> {
+  private readonly autocomplete = new AutocompleteProviderRegister();
+
   constructor(initializerContext: PluginInitializerContext) {}
 
   public setup(core: CoreSetup) {
-    return {};
+    return {
+      autocomplete: this.autocomplete,
+    };
   }
 
   public start(core: CoreStart) {}
```

Comment:
Setup needs to expose `getProvider` and `'addProvider`.
Start should to expose `getProvider`. 

Wherever you're using the provider in a react component, you should actually use `npStart`, as this is what they will generally use in NP. (I'm going to validate this with platform team)